src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs

```diff
@@ -27,36 +35,49 @@ public static class AdoNetIntegration
         /// <param name="mdToken">The mdToken of the original method call.</param>
         /// <returns>The value returned by the instrumented method.</returns>
         [InterceptMethod(
-            TargetAssembly = "System.Data", // .NET Framework
-            TargetType = "System.Data.Common.DbCommand",
-            TargetSignatureTypes = new[] { "System.Data.Common.DbDataReader", "System.Data.CommandBehavior" },
+            TargetAssembly = FrameworkAssembly,
+            TargetType = DbCommand,
+            TargetSignatureTypes = new[] { DbDataReader, CommandBehavior },
             TargetMinimumVersion = Major4,
             TargetMaximumVersion = Major4)]
         [InterceptMethod(
-            TargetAssembly = "System.Data.Common", // .NET Core
-            TargetType = "System.Data.Common.DbCommand",
-            TargetSignatureTypes = new[] { "System.Data.Common.DbDataReader", "System.Data.CommandBehavior" },
+            TargetAssembly = CoreAssembly,
+            TargetType = DbCommand,
+            TargetSignatureTypes = new[] { DbDataReader, CommandBehavior },
             TargetMinimumVersion = Major4,
             TargetMaximumVersion = Major4)]
         public static object ExecuteDbDataReader(object @this, int behavior, int opCode, int mdToken)
         {
             var command = (DbCommand)@this;
             var commandBehavior = (CommandBehavior)behavior;
+            var instanceType = command.GetType();
 
-            var executeReader = Emit.DynamicMethodBuilder<Func<object, CommandBehavior, object>>
-               .GetOrCreateMethodCallDelegate(
-                    command.GetType(),
-                    "ExecuteDbDataReader");
+            Func<object, CommandBehavior, object> instrumentedMethod = null;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Func<object, CommandBehavior, object>>
+                       .Start(instanceType.Assembly, mdToken, opCode, nameof(ExecuteDbDataReader))
```

Comment:
I still don't quite understand what was wrong with using `Assembly.GetCallingAssembly()` instead of `instanceType.Assembly`. Could you elaborate on why that didn't work?

Reply:
![image](https://user-images.githubusercontent.com/1801443/60983115-6a98ad00-a307-11e9-991a-c92f895eb58f.png)

Reply:
In retrospect, looks like we should've dug more into this one :)